src/textual/screen.py

```diff
@@ -783,7 +783,9 @@ def _refresh_layout(
         if self.is_current:
             self._compositor_refresh()
 
-        if not self.app._dom_ready:
+        if self.app._dom_ready:
```

Comment:
Does this method run when you scroll?

Reply:
In testing it very much appears to (see the test code linked to above, one of the test areas is a scrolling region).